### PR TITLE
feat: search locations in a different activity

### PR DESCRIPTION
**Pull request by nikit19**
First steps of https://github.com/fossasia/open-event-android/issues/169
gif:
ezgif com-video-to-gif 9

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_search_location)
+        this.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        this.supportActionBar?.title = ""
+
+        search.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                //Do your search
+                searchLocationViewModel.saveSearch(query)
+                val startMainActivity  = Intent(this@SearchLocationActivity, MainActivity::class.java)
```
[the remark this answers is not in the document]
Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt]: @nikit19 can't we use SearchLocationActivity.this instead?

[the remark this answers is not in the document]
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt]: Only 'this' might work too I guess @nikit19 @simarsingh24

[the remark this answers is not in the document]
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt]: Neither of the two works.
other than that I can make it applicationContext
that works

**Comment by nikit19** [comment]
@iamareebjamal Please review

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 2]
```diff
@@ -73,6 +73,9 @@ class EventsFragment : Fragment() {
             rootView.locationTextView.text = eventsViewModel.savedLocation
             eventsViewModel.loadLocationEvents(eventsViewModel.savedLocation.toString())
         }
+        else {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Style

**Comment by nikit19** [comment]
@iamareebjamal PR Updated Please review

**Pull request merged**